lib/active_stash/collection_proxy.rb

```diff
@@ -0,0 +1,78 @@
+require 'forwardable'
+
+module ActiveStash
+  class CollectionProxy
+    extend Forwardable
+    def_delegators :collection, :query
+    def_delegators :@model, :collection_name, :stash_indexes
+
+    def initialize(model)
+      @model = model
+    end
+
+    def upsert(stash_id, attrs, store_record:)
+      collection.upsert(stash_id, attrs, store_record: store_record)
+    rescue GRPC::NotFound
+      raise NoCollectionError, name: collection_name
+    end
+
+    def schema
+      SchemaBuilder.new(@model).build
+    end
+
+    def create!
+      client.create_collection(collection_name, schema)
+      logger.info("Successfully created '#{collection_name}'")
+      true
+    rescue GRPC::AlreadyExists
```

Comment:
Ditto for this (and all other instances where you've rescued GRPC::* from now on).